File: Basic/Math/CountDig.java

public class CountDig {
    public static int solve(int n) {
        int cnt = 0;
        while (n != 0) {
            n /= 10;
            cnt++;
        }
        return cnt;
    }

    public static int countDigitsLog(int n) {
        if (n == 0) {
            return 1;
        }
        return (int) (Math.log10(n) + 1);
    }
}

File: Basic/Math/CountDig.py
import math


def solve(number: int) -> int:
    count = 0
    while number != 0:
        number //= 10
        count += 1
    return count


def count_digits_log(number: int) -> int:
    if number == 0:
        return 1
    return int(math.log10(number) + 1)
